src/cephadm/cephadm

```diff
@@ -6515,10 +6521,21 @@ def command_rm_cluster(ctx):
             if ctx.fsid in f.read():
                 valid_fsid = True
         if valid_fsid:
+
+            # rm sysctl settings
+            sysctl_dir = Path(ctx.sysctl_dir)
+            for p in sysctl_dir.glob(f'90-ceph-{ctx.fsid}-*.conf'):
+                p.unlink()
+
+            # rm configuration files on /etc/ceph
             for n in range(0, len(files)):
                 if os.path.exists(files[n]):
                     os.remove(files[n])
 
+            # cleanup remaining ceph directories
+            ceph_dirs = [f'/run/ceph/{ctx.fsid}', f'/tmp/var/lib/ceph/{ctx.fsid}', f'/var/run/ceph/{ctx.fsid}']
+            for dd in ceph_dirs:
+                shutil.rmtree(dd, ignore_errors=True)
```

Comment:
I think the cleanup for the ceph dirs and sysctl dir should be moved outside the `if os.path.exists(files[0]):` block. That check is for if the `/etc/ceph/ceph.conf` file actually exists. That can be false and we can still have these `/run/ceph/<fsid>` etc. directories present (if the cluster was not named "ceph"). The way you're removing them and what you're removing lgtm I just think it shouldn't be restricted by that check. Was there some reason in your testing that made them have to be in this bock?

Reply:
I moved them inside because I though this code should only be executed if the value in /etc/ceph/ceph.conf is valid. If this is not the case then we can move this code outside the valid_fsid check. BTW: we are removing also the ceph.client.admin.keyring and ceph.pub. I'm not sure if this could affect other clusters!

Reply:
the check for the fsid being valid here is to check whether these **ceph.conf**, **ceph.client.admin.keyring** and **ceph.pub** belong to this cluster (by "this cluster" I mean the cluster with the given fsid) because in a multi-cluster enviroment they may not. However, these directories you're adding cleaning for here e.g. `/run/ceph/<fsid>` and the sysctl dir have the fsid in the name so we don't need to check for some match between the fsid for the cluster we're removing with these directoires. The directories must belong to the cluster with this fsid because otherwise they wouldn't have the fsid in their name.

Reply:
so the 3 files in /etc/ceph in particular need this check for that reason, but other things we're cleaning do not.

Reply:
Okay, thanks for the detailed explanation. I've just moved the code outside the valid_fsid check.